src/v2/providers/firestore.ts

```diff
@@ -95,6 +115,10 @@ export interface FirestoreEvent<T, Params = Record<string, string>> extends Clou
   namespace: string;
   /** The document path */
   document: string;
+  /** The type of principal that triggered the event */
+  authType?: AuthType;
```

Comment:
These will always be populated in the new event type, right? If so, we should create a separate event type where the fields are omitted in the previous event handlers and guaranteed to exist in the new ones.

Reply:
@exaby73 actually made the same suggestion when implementing the triggers in the Python SDK, but my reasoning for using the same event type is that we had discussed plans for aliasing `onDocumentX` to `onDocumentXWithAuthContext` in the next major release of the SDK, which may pollute the namespace a bit and cause confusion; a user writing a `onDocumentX` function will need to use the `FirestoreEventWithAuthContext` interface as opposed to just a single `FirestoreEvent` interface. The additional `authId` field is also optional and not guaranteed to exist in the delivered event.

But I also see the argument for having a separate event type. WDYT? Happy to make the changes and coordinate with @exaby73 to restore his old changes if we think the correct approach.

Reply:
I'm pretty strongly in favor of fields being non-optional when they will always be provided and omitted when they will never be provided (in TypeScript at least. I'm not the expert for Python)